src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -704,6 +704,21 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
             call_args_vec.p[arg_pos].loc = expr->base.loc;
             call_args_vec.p[arg_pos].m_value = expr;
         }
+        //Filling missing arguments with its default argument passed in function definition (if existed).
+        for(size_t i = 0; i < orig_func->n_args; i++ ){
+            if(call_args_vec.p[i].m_value == nullptr){
+                ASR::symbol_t* sym = ASR::down_cast<ASR::Var_t>(orig_func->m_args[i])->m_v;
+                std::string variable_name = ASR::down_cast<ASR::Variable_t>(sym)->m_name;
+                ASR::Variable_t* var = ASR::down_cast<ASR::Variable_t>(orig_func->m_symtab->get_symbol(variable_name));
+                if (var->m_symbolic_value == nullptr){
+                    call_args_vec.reserve(al, n_pos_args + n_kwargs); //deprecate Vec container, so it raise error later.
```

Comment:
Something is not right here --- why do you need to call `reserve` here?

Reply:
Right. I think we should better throw a `SemanticError` when a default value is expected/needed, but is `nullptr`.